src/lava/lib/optimization/solvers/generic/scif/models.py

```diff
@@ -16,145 +17,176 @@
-        sig_unsat_idx = np.where(spk_hist_buffer == 2)
-        sig_spk_idx = np.where(np.logical_and(spk_hist_buffer == 1,
-                                              self.cnstr_intg == 0))
-
-        # Assign sigma spikes (+/- 1)
-        s_sig[sig_unsat_idx] = -1
-        s_sig[sig_spk_idx] = 1
-
-        return s_sig
-
-    def _gen_wta_spks(self, spk_hist_buffer):
-        # Indices of WTA neurons signifying unsatisfied constraints, based on
-        # buffered history from previous timestep
-        wta_unsat_prev_ts_idx = np.where(np.logical_and(spk_hist_buffer == 1,
-                                                        self.cnstr_intg < 0))
-
-        # Reset voltages of unsatisfied WTA
-        self.state[wta_unsat_prev_ts_idx] = 0
-        # indices of neurons to be integrated:
-        intg_idx = np.where(self.state >= 0)
-        # indices of neurons in refractory:
-        rfct_idx = np.where(self.state < 0)
+    # This method is overloaded for CSP and QUBO
+    def _get_local_validity_conflict(self, spk_hist_status):
+        return np.ones_like(self.cnstr_intg).astype(bool), np.zeros_like(
+            self.cnstr_intg).astype(bool)
 
-        # Indices of WTA neurons that will spike and enter refractory
-        wta_spk_idx = self._integration_dynamics(intg_idx)
-
-        # Indices of WTA neurons coming out of refractory or those signifying
-        # unsatisfied constraints
-        wta_rfct_end_or_unsat_idx = self._refractory_dynamics(rfct_idx) if \
-            self.neg_tau_ref != 0 else (np.array([], dtype=np.int32),)
-
-        s_wta = np.zeros_like(self.state)
-        s_wta[wta_spk_idx] = 1
-        s_wta[wta_unsat_prev_ts_idx] = -1
-        s_wta[wta_rfct_end_or_unsat_idx] = -1
-
-        return s_wta
+    # This method is overloaded for CSP and QUBO
+    def _gen_sig_spks(self, spk_hist_status, local_validity):
+        return np.zeros_like(self.state)
 
     def _integration_dynamics(self, intg_idx):
         state_to_intg = self.state[intg_idx]  # voltages to be integrated
         cnstr_to_intg = self.cnstr_intg[intg_idx]  # currents to be integrated
-        spk_hist_to_intg = self.spk_hist[intg_idx]  # beta to be integrated
         step_size_to_intg = self.step_size[intg_idx]  # bias to be integrated
 
-        lfsr = self._prng(intg_idx_=intg_idx)
-
-        state_to_intg = state_to_intg + lfsr + cnstr_to_intg + step_size_to_intg
-        np.clip(state_to_intg, a_min=0, a_max=2 ** 23 - 1, out=state_to_intg)
-
-        # WTA spike indices when threshold is exceeded
-        wta_spk_idx = np.where(state_to_intg >= self.theta)  # Exceeds threshold
+        lfsr = self._prng()
+        lfsr_to_intg = lfsr[intg_idx]
 
-        # Spiking neuron voltages go in refractory (if neg_tau_ref < 0)
-        state_to_intg[wta_spk_idx] = self.neg_tau_ref  # Post spk refractory
-        spk_hist_to_intg[wta_spk_idx] |= 1
+        state_to_intg = state_to_intg + lfsr_to_intg + cnstr_to_intg + \
```

Comment:
Did we really say that the step size and the (purely postivie) lfsr are supposed to be additive factors? Did you write down the precise dynamic equstions somewhere, maybe including the rational? Would appreciate to have a look.